src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj

```diff
@@ -78,4 +78,23 @@
     </ProjectReference>
   </ItemGroup>
 
+  <Target Name="GenerateBuildMetadata" BeforeTargets="BeforeBuild">
+    <WriteLinesToFile
+      File="$(IntermediateOutputPath)Generated_BuildMetadata.cs"
+      Lines="// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+internal static class BuildMetadata
+{
+    public const string NetCoreAppCurrentVersion = &quot;$(NetCoreAppCurrentVersion)&quot;%3B
+    public const string NetCoreAppToolCurrentVersion = &quot;$(NetCoreAppToolCurrentVersion)&quot;%3B
```

Comment:
It would be better to name this specific to why we need it, i.e. ILLinkTargetFramework and pass `NetCoreAppToolCurrent` in.

Reply:
Yup, if we want to make this target general purpose, then the names can be transparent / agnostic of consumer's context. Otherwise, if we are going to keep it to only this project, makes sense to name them accordingly.